overlord/snapstate/backend/link.go

```diff
@@ -68,6 +68,33 @@ type LinkContext struct {
 	// SkipBinaries indicates that we should skip removing snap binaries,
 	// icons and desktop files in UnlinkSnap
 	SkipBinaries bool
+
+	// HasOtherInstances indicates that other instances of the snap are
+	// already installed in the system.
+	HasOtherInstances bool
+}
+
+func createSharedSnapDirForParallelInstance(s snap.PlaceInfo) error {
+	_, key := snap.SplitInstanceName(s.InstanceName())
+
+	if key != "" {
+		err := os.MkdirAll(snap.BaseDir(s.SnapName()), 0755)
+		if err != nil && !os.IsExist(err) {
+			return err
+		}
+	}
+	return nil
+}
+
+func removeSharedSnapDirForParallelInstance(s snap.PlaceInfo) error {
+	_, instanceKey := snap.SplitInstanceName(s.InstanceName())
+
+	if instanceKey != "" {
+		// remove only if not used by other instances of the same snap,
+		// failure to remove is ok, there may be other revisions
+		os.Remove(snap.BaseDir(s.SnapName()))
+	}
+	return nil
```

Comment:
we can remove the error here
```suggestion
func removeSharedSnapDirForParallelInstance(s snap.PlaceInfo) {
	_, instanceKey := snap.SplitInstanceName(s.InstanceName())

	if instanceKey != "" {
		// remove only if not used by other instances of the same snap,
		// failure to remove is ok, there may be other revisions
		os.Remove(snap.BaseDir(s.SnapName()))
	}
```